plugins/azure-devops-backend/api-report.md

```diff
@@ -23,12 +25,31 @@ export class AzureDevOpsApi {
   // (undocumented)
   getAllTeams(): Promise<Team[]>;
   // (undocumented)
+  getBuildDefinitions(
+    projectName: string,
+    definitionName: string,
+  ): Promise<BuildDefinitionReference[]>;
+  // (undocumented)
   getBuildList(
     projectName: string,
     repoId: string,
     top: number,
   ): Promise<Build[]>;
   // (undocumented)
+  getBuildRuns(
+    projectName: string,
+    top: number,
+    repoName?: string,
+    definitionName?: string,
+  ): Promise<BuildRun[]>;
```

Comment:
No need to change this right now, but as a general hint: We have found it to be a useful pattern, especially in broad APIs like this, to pass in a single object as a parameter instead of many individual parameters. Example:

```ts
  getBuildRuns(
    options: {
      projectName: string,
      top: number,
      repoName?: string,
      definitionName?: string,
    }
  ): Promise<BuildRun[]>;
```

The outcome is that it's way easier to evolve (adding more parameters in the options structure, making things optional etc) without completely breaking the interface. You sidestep annoying pitfalls with the many-argument situation, like the impossibility of adding more arguments that are mandatory, _after_ non-mandatory ones. You also make the call sites look a lot prettier because of the named explicit object keys, so they aren't all like `.callTheThing('foo', true, true, 5, 'bar')` without any clarity to the reader what they individually mean.

I guess the azure API client itself serves as a good example, with its endless lists of `undefined` arguments, why this approach can be nice. :)

Again, just a hint for the future, feel free to ignore :)